foundry/core/redux_store.py

```diff
@@ -0,0 +1,90 @@
+from typing import TypeVar, Generic
+import copy
+from dataclasses import dataclass
+from abc import ABC, abstractmethod
+
+@dataclass
+class Action:
+    """ User action that will be dispatched to the store for processing
+    
+    type: str - a unique id that defines what action is happening
+    payload: any - the data payload of the action, can be any type.
+    """
+    type: str
+    payload: any
+
+S = TypeVar('S')
+
+class ReduxStore(ABC, Generic[S]):
+    """ An abstract Redux store for handing system state and user actions.
+
+    Redux is a pattern used for handing system state storage/change based on
+    user interation.  More information can be found here:
+    https://redux.js.org/introduction/core-concepts
+
+    The basic principle is that the store hold the system state and accepts
+    user actions and creates a new system state based on that action.
+
+    Actions are dispatched to the store and the store will notify all
+    subscribers if the state changes.
+
+    In this implementation the state is a generic type 'S' to be defined by
+    the developer.
+    """
+    _default_state: S = None
+    _state: S = None
```

Comment:
`_state` is promised to be a generic type; however, you have set it to None. Either `_state` should be an optional generic or a property should be created that should be created to raise an exception if the state is None.